C7GameData/Tile.cs

```diff
@@ -79,17 +79,22 @@ public Tile()
 		//Sometimes we care *specifically* about the Coast terrain, e.g. galleys can only move on that terrain, not Sea or Ocean
 		//Those cases should not use this method.
 		public bool NeighborsWater() {
-			foreach (Tile neighbor in getDiagonalNeighbors()) {
+			foreach (Tile neighbor in neighbors.Values) {
```

Comment:
Remember that cities are a special case for diagonal neighbors. Otherwise a diagonal water neighbor only is not considered connected, ie the land connection takes priority. Is this taken into account when there is no city?

Reply:
Do you mean in terms of trade connections?  I guess I'm not sure exactly what case it is you mean where a diagonal (vertex) neighbor doesn't count for water, other than the graphics case mentioned in the comment.

So far this is only used when deciding what a city can build, whether a barbarian camp spawns boats, and how appealing a city site is to the AI.

Reply:
If you have water tiles at N/S and land tiles at E/W, that's an isthmus and sea units can't move between the water tiles. But if N is a city, sea units can move in and out of it. I just want to be sure this method isn't a general case for finding connected water tiles.

Reply:
Ohhhh, right!  Forgot about that.  Well, the isthmus case _is_ a problem, but it doesn't use this method.  For both barbarian and non-barbarian AIs, the `GetCoastNeighbors` method in `Tile.cs` is used, which doesn't check for isthmuses.  I'm not sure off the top of my head for human moves.  But I have seen AI galleys move across isthmuses.

I don't think this method could be easily misused for that, as it returns a boolean, and our movement/pathing methods are looking for tile objects that can be moved to.  I.e. this method doesn't tell you which neighboring tiles are water, so you can't use it to decide where to move.

Nonetheless, that is a good catch, I wrote it up as #369 .  I'd probably do it after #148 , which I see as the next iteration in improving the accuracy of movement possibilities.